collector/__init__.py

```diff
@@ -114,61 +114,51 @@ def __init__(self,
-
-        xmlselect = [site_name.upper() + '-' + sem + '-' + prog_type
-                     for sem in self.semesters for prog_type in self.program_types]
-
-        # Retrieve the site details
-        try:
-            site = Site(site_name)
-            # TODO: Possibly get rid of these members from Collector and just access them directly?
-            self.locations[site] = GEOGRAPHICAL_LOCATIONS[site]
-            self.timezones[site] = TIME_ZONES[site]
-        except ValueError:
-            raise RuntimeError(f'{site_name} not a valid site name.')
-        except astropy.coordinates.UnknownSiteException:
-            raise RuntimeError(f'{site_name} does not correspond to a valid geographical location.')
-        except pytz.exceptions.UnknownTimeZoneError:
-            raise RuntimeError(f'{site_name} is not associated with a known timezone.')
-
-        self._calculate_night_events()
-
-        # TODO: We will have to modify in order for this code to be usable by other observatories.
-        zip_path = os.path.join(path,
-                                f'{(self.time_range[0] - 1.0 * u.day).strftime("%Y%m%d")}{SITE_ZIP_EXTENSIONS[site]}')
-        logging.info(f'Retrieving program data from: {zip_path}.')
-
-        time_accounting = self._load_tas(path, site)
-        self._readzip(zip_path, xmlselect, site_name, tas=time_accounting, obsclasses=self.obs_classes)
+        
+        
+        for site in self.sites:
+            site_name = site.value
+
+            xmlselect = [site_name.upper() + '-' + sem + '-' + prog_type
+                        for sem in self.semesters for prog_type in self.program_types]
+
+            # Retrieve the site details
+            try:
+                #site = Site(site_name)
+                # TODO: Possibly get rid of these members from Collector and just access them directly?
+                self.locations[site] = GEOGRAPHICAL_LOCATIONS[site]
+                self.timezones[site] = TIME_ZONES[site]
+            except ValueError:
+                raise RuntimeError(f'{site_name} not a valid site name.')
+            except astropy.coordinates.UnknownSiteException:
+                raise RuntimeError(f'{site_name} does not correspond to a valid geographical location.')
+            except pytz.exceptions.UnknownTimeZoneError:
+                raise RuntimeError(f'{site_name} is not associated with a known timezone.')
+
+            # TODO: We will have to modify in order for this code to be usable by other observatories.
+            zip_path = os.path.join(path,
+                                    f'{(self.time_range[0] - 1.0 * u.day).strftime("%Y%m%d")}{SITE_ZIP_EXTENSIONS[site]}')
+            logging.info(f'Retrieving program data from: {zip_path}.')
+
+            time_accounting = self._load_tas(path, site)
+            self._calculate_night_events(site)
+            self._readzip(zip_path, xmlselect, site, tas=time_accounting)
+            Collector.observation_num = 0
```

Comment:
I guess we set `Collector.observation_num` here to 0 because we are starting over for a different site?
Since this is never used outside of `Collector._process_observation_data`, I think we should eliminate it from being a member of `Collector` and just moving `observation_num` into `_process_observation_data`.

I think we can put it right after the new L132 and just write:
```
observation_num = 0
```

Reply:
Upon further investigation, maybe this is not so simple, but I would like to get this PR merged and then see if we can get rid of the `Collector.observation_num` variable entirely. I'll take a look at this after.